org/lateralgm/subframes/FontFrame.java

```diff
@@ -410,122 +433,100 @@ public void commitChanges()
 		res.setName(name.getText());
 		}
 
-	public void actionPerformed(ActionEvent ev)
+	private void deleteSelectedRange()
 		{
-		String com = ev.getActionCommand();
-		if (com.equals("Normal")) //$NON-NLS-1$
+		int sel = rangeList.getSelectedIndex();
+		if (sel != -1)
+			res.characterRanges.remove(sel);
+		if (sel >= res.characterRanges.size())
 			{
-			CharacterRange cr = rangeList.getSelectedValue();
-			if (cr != null)
-				{
-				cr.properties.put(PCharacterRange.RANGE_MIN,32);
-				cr.properties.put(PCharacterRange.RANGE_MAX,127);
-				}
-			return;
+			if (res.characterRanges.isEmpty())
+				rangeList.clearSelection();
+			else
+				rangeList.setSelectedIndex(res.characterRanges.size() - 1);
 			}
-		else if (com.equals("ASCII")) //$NON-NLS-1$
-			{
-			CharacterRange cr = rangeList.getSelectedValue();
-			if (cr != null)
-				{
-				cr.properties.put(PCharacterRange.RANGE_MIN,0);
-				cr.properties.put(PCharacterRange.RANGE_MAX,255);
-				}
-			return;
-			}
-		else if (com.equals("Digits")) //$NON-NLS-1$
-			{
-			CharacterRange cr = rangeList.getSelectedValue();
-			if (cr != null)
-				{
-				cr.properties.put(PCharacterRange.RANGE_MIN,48);
-				cr.properties.put(PCharacterRange.RANGE_MAX,57);
-				}
-			return;
-			}
-		else if (com.equals("Letters")) //$NON-NLS-1$
-			{
-			CharacterRange cr = rangeList.getSelectedValue();
-			if (cr != null)
-				{
-				cr.properties.put(PCharacterRange.RANGE_MIN,65);
-				cr.properties.put(PCharacterRange.RANGE_MAX,122);
-				}
-			return;
-			}
-		else if (com.equals("FromPreview"))
+		else
+			fireRangeUpdate();
+		}
+
+	private void setRange(int min, int max)
+		{
+		CharacterRange cr = rangeList.getSelectedValue();
+		if (cr != null)
 			{
-			res.addRangesFromString(previewText.getText());
-			return;
+			cr.properties.put(PCharacterRange.RANGE_MIN, min);
+			cr.properties.put(PCharacterRange.RANGE_MAX, max);
 			}
-		else if (com.equals("FromString"))
+		}
+
+	public void actionPerformed(ActionEvent ev)
+		{
+		switch (ev.getActionCommand())
 			{
-			String result = JOptionPane.showInputDialog(this,"","Character Sequence",
-					JOptionPane.PLAIN_MESSAGE);
-			if (result != null)
+			case "Normal": //$NON-NLS-1$
+				setRange(32, 127);
+				break;
+			case "ASCII": //$NON-NLS-1$
+				setRange(0, 255);
+				break;
+			case "Digits": //$NON-NLS-1$
+				setRange('0', '9');
+				break;
+			case "Letters": //$NON-NLS-1$
+				setRange('A', 'z');
+				break;
+			case "FromPreview":
```

Comment:
You left off some internationalization tags. Does the build system not enforce that, anymore? I recall it being an Eclipse warning. I guess this broke down long ago, because the file was already missing some...

Reply:
This is easy to correct if that's the case Josh.

I believe many of the ones in this file may have come from me. I turned off those warnings a while back when I was reducing the warning count. In order to see the actual warnings that mattered, I had to filter out these warnings. Obviously, I wasn't planning to fix all of the NON-NLS warnings because that would have polluted my diff. We can look at reenabling the warnings and make a single giant commit to address all of those NON-NLS warnings at once.

Regardless, for any new lines he's added here he should definitely double check that he's added the NON-NLS comments. I'm about to check this out myself so I'll probably just do it for him.

Reply:
I just checked locally and no they are still enabled, so we're all good. Everybody has just been ignoring them because they are mainly for translators. Like I said, I've been turning them off to focus on the important warnings.
![Unlocalized Warnings](https://user-images.githubusercontent.com/3212801/56187711-87b42d00-5ff1-11e9-84bd-33c152556714.png)